src/client/components/MealWithId.js

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router";
+import AddReservation from "./AddReservation";
+import AddReview from "./AddReview";
+import LoadImage from "../LoadImage";
+import Displaymeal from "./Displaymeal";
+import { useHistory } from 'react-router'
+
+const MealWithId = () => {
+  const [mealWithId, setMealWithId] = useState([]);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [reservationForm, setReservationForm] = useState(false);
+  const [reviewForm, setReviewForm] = useState(false);
+  const [reservation, setReservation] = useState([]);
+  const paramMealId = useParams();
+  const history = useHistory()
+  useEffect(() => {
+    fetch(`/api/meals?availableReservations=true`)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          throw new Error("Something went wrong!");
+        }
+      })
+      .then((data) => {
+        setReservation(data);
```

Comment:
This is mealsWithAvailableReservations, rather than reservations right?
I'd suggest you use that name, since it makes more sense :)